salt/client/__init__.py

```diff
@@ -1137,6 +1140,13 @@ def get_iter_returns(
                 minions = {minions}
             elif isinstance(minions, (list, tuple)):
                 minions = set(list(minions))
+        expected_syndics = set()
+        if self.opts["order_masters"]:
+            if not isinstance(kwargs["syndics"], set):
+                if isinstance(kwargs["syndics"], str):
+                    expected_syndics = {kwargs["syndics"]}
+                elif isinstance(kwargs["syndics"], (list, tuple)):
+                    expected_syndics = set(list(kwargs["syndics"]))
```

Comment:
**question**: Is this the best way to get the behavior here in the command line client? I *think* so. Though there may also still be some other legacy syndic specific code in here...